src/java/org/apache/cassandra/db/compaction/unified/AdaptiveController.java

```diff
@@ -135,19 +145,77 @@ static Controller fromOptions(Environment env,
                                   Map<String, String> options)
     {
         int scalingParameter = options.containsKey(STARTING_SCALING_PARAMETER) ? Integer.parseInt(options.get(STARTING_SCALING_PARAMETER)) : DEFAULT_STARTING_SCALING_PARAMETER;
-        int[] scalingParameters = new int[32];
-        int[] previousScalingParameters = new int[32];
-        //set the scaling parameter for each level to the starting scaling parameter or default scaling parameter
-        Arrays.fill(scalingParameters, scalingParameter);
-        Arrays.fill(previousScalingParameters, scalingParameter);
+        int[] scalingParameters = new int[UnifiedCompactionStrategy.MAX_LEVELS];
```

Comment:
Cassandra (and we as its developers) is quite sensible to allocations. Although it doesn't matter here as we'll only initialize the controller once, these allocations are wasted work in one of the two paths and can be moved to the place we need them.